### ISSUE TITLE
service crashes after a certain time

### ISSUE THREAD

**Issue by KalvinVilla** [SA-EME/SWA#11]
```
Nom de l’application défaillante SWA.Core.exe, version : 0.1.2.17, horodatage : 0x671a092e
Nom du module défaillant : KERNELBASE.dll, version : 10.0.17763.6414, horodatage : 0x4c9061ed
Code d’exception : 0xe0434352
Décalage d’erreur : 0x00126362
ID du processus défaillant : 0x2cf8
Heure de début de l’application défaillante : 0x01db2602f6fc4fa7
Chemin d’accès de l’application défaillante : C:\Program Files\SWA\SWA.Core.exe
Chemin d’accès du module défaillant: C:\Windows\System32\KERNELBASE.dll
ID de rapport : c28c138a-edbc-48de-bbc4-8d0eb6abde09
Nom complet du package défaillant : 
ID de l’application relative au package défaillant : 
```

```
Application : SWA.Core.exe
Version du Framework : v4.0.30319
Description : le processus a été arrêté en raison d'une exception non gérée.
Informations sur l'exception : System.InvalidOperationException
   à System.Collections.Generic.LinkedList`1+Enumerator[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].MoveNext()
   à System.Linq.Enumerable.Last[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.Collections.Generic.IEnumerable`1<System.__Canon>)
   à SWA.Core.Logs.LogListener.Event_NewLogWritten(System.Object, System.Diagnostics.Eventing.Reader.EventRecordWrittenEventArgs)
   à System.Diagnostics.Eventing.Reader.EventLogWatcher.IssueCallback(System.Diagnostics.Eventing.Reader.EventRecordWrittenEventArgs)
   à System.Diagnostics.Eventing.Reader.EventLogWatcher.HandleEventsRequestCompletion()
   à System.Diagnostics.Eventing.Reader.EventLogWatcher.RequestEvents()
   à System.Diagnostics.Eventing.Reader.EventLogWatcher.SubscribedEventsAvailableCallback(System.Object, Boolean)
   à System.Threading._ThreadPoolWaitOrTimerCallback.WaitOrTimerCallback_Context(System.Object, Boolean)
   à System.Threading._ThreadPoolWaitOrTimerCallback.WaitOrTimerCallback_Context_f(System.Object)
   à System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   à System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   à System.Threading._ThreadPoolWaitOrTimerCallback.PerformWaitOrTimerCallback(System.Object, Boolean)
```

**Comment by KalvinVilla** [comment]
the bug is due to a problem writing to a variable at the same time, we need to use a locker on the variable

**Comment by KalvinVilla** [comment]
fixed by [380ccdf](https://github.com/SA-EME/SWA/commit/380ccdfba14b798d8712247e20ac6ef724dbe58c)
